content/concepts/did-ddo.md

```diff
@@ -24,8 +24,17 @@ DIDs and DDOs follow [this specification](https://w3c-ccg.github.io/did-spec/) d
 - An asset should have a DID and DDO. The DDO should include metadata about the asset.
 - The DDO can only can be modified by _owners_ or _delegated users_.
 - There _must_ be at least one client library acting as _resolver_, to get a DDO from a DID.
-- The DDO is stored on-chain. It's stored in in plaintext, with two exceptions: (1) the field for resource-access url is encrypted (2) the whole DDO may be encrypted, if the publisher is willing to lose 100% of discoverability.
-- A metadata cache like Aquarius can help in reading and writing DDO data from the chain.
+- A metadata cache like Aquarius can help in reading and searching through DDO data from the chain.
+
+## Flow for publishing / retrieving DDOs
+
+- The DDO is stored on-chain. 
+- It's stored encrypted (using the private key of the provider). To resolve it, you must query the provider and you will might get the clear text ddo (depends on access rights, state, etc)
+- Each DID has a state, which is held by the NFT Contract (also stored in the DDO.status.status).  The possible states are:
```

Comment:
the did has the state? not the ddo/asset/nft? Isn't the did just an identifier of something.